index.js

```diff
@@ -1,83 +1,70 @@
-#!/usr/bin/env node
-
 var proc = require('child_process')
 var execspawn = require('execspawn')
 var os = require('os')
 var path = require('path')
 var fs = require('fs')
-var minimist = require('minimist')
 var abi = require('node-abi')
 var mkdirp = require('mkdirp')
 var xtend = require('xtend/immutable')
 
-var argv = minimist(process.argv.slice(2), {
-  alias: {target: 't', version: 'v', all: 'a'},
-  boolean: ['quiet', 'strip']
-})
-
-var arch = argv.arch || os.arch()
-var platform = argv.platform || os.platform()
-var cwd = argv._[0] || '.'
-var env = xtend(process.env, {ARCH: arch, PREBUILD_ARCH: arch})
-var builds = path.join(cwd, 'prebuilds', platform + '-' + arch)
-var output = path.join(cwd, 'build', argv.debug ? 'Debug' : 'Release')
+function prebuildify (opts, cb) {
+  opts = xtend({
+    arch: os.arch(),
+    platform: os.platform(),
+    cwd: '.',
+    targets: []
+  }, opts)
 
-var targets = [].concat(argv.target || []).map(function (v) {
-  if (v.indexOf('@') === -1) v = 'node@' + v
-
-  return {
-    runtime: v.split('@')[0],
-    target: v.split('@')[1].replace(/^v/, '')
+  if (!opts.targets.length) {
+    return cb(new Error('You must specify at least one target using --target=runtime@version'))
   }
-})
 
-// TODO: also support --lts and get versions from travis
-if (argv.all) {
-  targets = abi.supportedTargets.slice(0)
-}
+  if (!fs.existsSync(path.join(opts.cwd, 'package.json'))) {
+    return cb(new Error('No package.json found'))
+  }
 
-if (!targets.length) {
-  console.error('You must specify at least one target using --target=runtime@version')
-  process.exit(1)
-}
+  opts = xtend(opts, {
+    targets: opts.targets.slice(),
+    env: xtend(process.env, {ARCH: opts.arch, PREBUILD_ARCH: opts.arch}),
+    builds: path.join(opts.cwd, 'prebuilds', opts.platform + '-' + opts.arch),
+    output: path.join(opts.cwd, 'build', opts.debug ? 'Debug' : 'Release')
+  })
 
-if (!fs.existsSync(path.join(cwd, 'package.json'))) {
-  console.error('No package.json found')
-  process.exit(1)
+  mkdirp.sync(opts.builds)
```

Comment:
we should use the async api since this is a js api now

``` js
mkdirp(opts.builds, function (err) {
  if (err) return cb(err)
  loop(opts, cb)
})
```